view/frontend/templates/cart/cart-normalization.phtml

```diff
@@ -4,11 +4,11 @@
  * See Extend-COPYING.txt for license details.
  */
 $viewModel = $this->getData('viewModel');
-$shouldBalanceCart = $viewModel->isCartBalancingEnabled();
+$shouldBalanceCart = $viewModel->isCartBalancingEnabled() && $viewModel->isExtendProductProtectionEnabled();
```

Comment:
Ok to add this extra gate but I don't think technically we would ever run into a situation where cart balancing would get triggered without product protection being enabled.